corehq/apps/reports/standard/users/reports.py

```diff
@@ -147,48 +150,65 @@ def rows(self):
+            record.details['changed_via'],
+            record.message,
+            self._user_history_details_cell(record.details['changes'], domain),
+            ServerTime(record.changed_at).user_time(timezone).ui_string(USER_DATETIME_FORMAT),
+        ]
 
+    def _user_history_details_cell(self, changes, domain):
+        def _html_list(changes, unstyled=True):
+            items = []
+            for key, value in changes.items():
+                if isinstance(value, dict):
+                    value = _html_list(value, unstyled=unstyled)
+                elif isinstance(value, list):
+                    value = format_html(", ".join(value))
+                else:
+                    value = format_html(str(value))
+                items.append("<li>{}: {}</li>".format(key, value))
+
+            class_attr = "class='list-unstyled'" if unstyled else ""
+            return mark_safe(f"<ul {class_attr}>{''.join(items)}</ul>")
+
+        properties = UserHistoryReport.get_primary_properties(domain)
+        properties.pop("user_data", None)
+        primary_changes = {}
+        all_changes = {}
 
-def _user_history_details_cell(changes, domain):
-    def _html_list(changes, unstyled=True):
-        items = []
         for key, value in changes.items():
-            if isinstance(value, dict):
-                value = _html_list(value, unstyled=unstyled)
-            elif isinstance(value, list):
-                value = format_html(", ".join(value))
-            else:
-                value = format_html(str(value))
-            items.append("<li>{}: {}</li>".format(key, value))
-
-        class_attr = "class='list-unstyled'" if unstyled else ""
-        return mark_safe(f"<ul {class_attr}>{''.join(items)}</ul>")
-
-    properties = UserHistoryReport.get_primary_properties(domain)
-    properties.pop("user_data", None)
-    primary_changes = {
-        properties.get(key, key): value for key, value in changes.items()
-        if key in properties
-    }
-    more_count = len(changes) - len(primary_changes)
-
-    return render_to_string("reports/standard/partials/user_history_changes.html", {
-        "primary_changes": _html_list(primary_changes) if primary_changes else None,
-        "all_changes": _html_list(changes, unstyled=False),
-        "more_count": more_count,
+            if key in properties:
+                if key in ('commcare_location_id', 'location_id'):
+                    value = self._get_location_name(value)
+                primary_changes[properties[key]] = value
+                all_changes[properties[key]] = value
+            if key == 'user_data':
+                for key, value in changes['user_data'].items():
+                    all_changes[key] = value
+
+        more_count = len(all_changes) - len(primary_changes)
+        return render_to_string("reports/standard/partials/user_history_changes.html", {
+            "primary_changes": _html_list(primary_changes) if primary_changes else None,
```

Comment:
nitpick: these two lines would read more smoothly if `_html_list` could accept `None` and just returned `None`